src/android/InAppChromeClient.java

```diff
@@ -100,39 +104,66 @@ public void onGeolocationPermissionsShowPrompt(String origin, Callback callback)
     @Override
     public boolean onJsPrompt(WebView view, String url, String message, String defaultValue, JsPromptResult result) {
         // See if the prompt string uses the 'gap-iab' protocol. If so, the remainder should be the id of a callback to execute.
-        if (defaultValue != null && defaultValue.startsWith("gap")) {
-            if(defaultValue.startsWith("gap-iab://")) {
-                PluginResult scriptResult;
-                String scriptCallbackId = defaultValue.substring(10);
-                if (scriptCallbackId.matches("^InAppBrowser[0-9]{1,10}$")) {
-                    if(message == null || message.length() == 0) {
-                        scriptResult = new PluginResult(PluginResult.Status.OK, new JSONArray());
-                    } else {
-                        try {
-                            scriptResult = new PluginResult(PluginResult.Status.OK, new JSONArray(message));
```

Comment:
Wow the nesting :)

Reply:
IKR, this is why I argue so often in favour of guard clauses.

An older (even than me) dev explained why you see code like this. Back in Þe olden times editors didn't have syntax highlighting. Especially in languages like c (predominant at the time) knowing where you returned out was especially important so people got into the habit of only returning once, at the bottom of the method. This meant you typically declared any state at the top, and set it in the logic. This makes guard clauses "impossible" as they rely on returning out as soon as you know the value. In fact my refactor would be seen as a retrograde step if you were a c-programmer using vi in the 80s/90s.

So what you see here is the code of someone who was probably taught (like I was) by people who'd used C or C-derivatives to code this way because it is better. Though interestingly they've ditched the single return point.